Mage.Sets/src/mage/cards/s/SanctumOfAll.java

```diff
@@ -0,0 +1,174 @@
+package mage.cards.s;
+
+import mage.abilities.Ability;
+import mage.abilities.common.BeginningOfUpkeepTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.dynamicvalue.common.PermanentsOnBattlefieldCount;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.hint.ValueHint;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.constants.TargetController;
+import mage.constants.Zone;
+import mage.filter.FilterCard;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterControlledPermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.Target;
+import mage.target.common.TargetCardInLibrary;
+import mage.target.common.TargetCardInYourGraveyard;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class SanctumOfAll extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterControlledPermanent();
+    static final PermanentsOnBattlefieldCount count = new PermanentsOnBattlefieldCount(filter);
+
+    static {
+        filter.add(SubType.SHRINE.getPredicate());
+    }
+
+    public SanctumOfAll(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{W}{U}{B}{R}{G}");
+        
+        this.addSuperType(SuperType.LEGENDARY);
+        this.subtype.add(SubType.SHRINE);
+
+        // At the beginning of your upkeep, you may search your library and/or graveyard for a Shrine card and put it onto the battlefield. If you search your library this way, shuffle it.
+        this.addAbility(new BeginningOfUpkeepTriggeredAbility(new SanctumOfAllSearchEffect(), TargetController.YOU, true));
+
+        // If an ability of another Shrine you control triggers while you control six or more Shrines, that ability triggers an additional time.
+        this.addAbility(new SimpleStaticAbility(new SanctumOfAllTriggerEffect()).addHint(new ValueHint("Shrines you control", count)));
```

Comment:
This should only copy triggered effects right? Does not copy the activated abilities of the white or red shrines.